src/components/activities/searchPatientActivity/SearchPatientActivity.tsx

```diff
@@ -32,19 +33,31 @@ const SearchPatientActivity: FunctionComponent<TProps> = ({
   };
 
   const validationSchema = object({
-    id: string().required("This field is required"),
+    //id: string().required("This field is required"),
     //TODO: write schema
   });
 
   const formik = useFormik({
     initialValues,
     validationSchema,
-    onSubmit: () => {
-      //TODO: implement onSubmit event
-      console.log("submit");
+    onSubmit: (values: IValues) => {
+      searchPatient(values);
     },
   });
 
+  const searchPatient = (values: IValues) => {
+    //TODO: if there is at least one of those fields Tax Number and Patient ID filled up, use getPatientUsingGET
+    const api = new PatientControllerApi();
```

Comment:
We don't perform API calls directly from inside the component. We use **Redux** for handling the state. Take a look at how it is done on the LoginActivity (`src/state/main`)